Modules/_localemodule.c

```diff
@@ -422,8 +422,10 @@ _locale_strxfrm_impl(PyObject *module, PyObject *str)
         goto exit;
     }
     if (n2 >= (size_t)n1) {
-        /* more space needed */
-        wchar_t * new_buf = PyMem_Realloc(buf, (n2+1)*sizeof(wchar_t));
+        /* more space needed, some implementations return needed size while
+           others just buffer length and it's up to the caller to figure
+           out needed buffer size */
+        wchar_t * new_buf = PyMem_Realloc(buf, n2 * 2 * sizeof(wchar_t));
```

Comment:
What's the point of allocating more memory if you still pass `n2+1` to wcsxfrm() below? I don't understand how this fix works :-)

Reply:
Sorry, I am still on 14.5, so I cannot check :(

Reply:
The fix comes from: https://github.com/python/cpython/issues/130567#issuecomment-2684536704

> They figured out that wcsxfrm does not actually return desired string size in case of the failure, so they ended up allocating double space in that case, see https://github.com/SWI-Prolog/swipl-devel/commit/f0bbfb082985382d777c3cfb84fc85e7e0ae345f.

But, it looks like that "double the memory is good enough".

Reply:
> What's the point of allocating more memory if you still pass `n2+1` to wcsxfrm() below? I don't understand how this fix works :-)

It apparently doesn't. Fixing that now. I was hoping for macos-15 being in CI because I don't have access to it either.

Reply:
> But, it looks like that "double the memory is good enough".

Well, it seems to cover most of the cases. They are doing the doubling the size in the loop until it succeeds. That seemed too aggressive to me.